unicornherder/herder.py

```diff
@@ -260,10 +260,28 @@ def _emergency_slaughter():
             pass
 
 
-def _wait_for_workers(overlap):
-    # TODO: do something smarter here
-    time.sleep(overlap)
-
+def _wait_for_workers(overlap, new_process, old_process):
+    # We expect the current process has one extra child (the new process that
+    # was forked aside from the usual number of workers
+    current_workers = len(old_process.children()) - 1
+    # We hope for same number of workers, if we don't have that we'll accept 1
+    expected_children = current_workers or 1
```

Comment:
It seems like this is only relevant if `current_workers` is set to `0`? What situation do we anticipate that happening, is it if the old process died unexpectedly?

And if so, I wonder if we should do `max(current_workers, 1)` to catch a situation where `len(old_process.children())` is `0` and therefore `current_workers` is `-1`?

Reply:
That sounds a good idea, it was mostly just defensive as waiting for anything less than one didn't seem logical.